PhotoDetailScreen.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect } from 'react';
+import { StyleSheet, View, Text, Image, TouchableOpacity } from 'react-native';
+import { RouteProp, useRoute, useNavigation } from '@react-navigation/native';
+
+type RootStackParamList = {
+  PhotoDetail: { imageUrl: string };
+  ImageModal: { imageUrl: string };
+};
+
+type PhotoDetailRouteProp = RouteProp<RootStackParamList, 'PhotoDetail'>;
+
+const PhotoDetailScreen = () => {
+  const route = useRoute<PhotoDetailRouteProp>();
+  const imageUrl = route.params.imageUrl;
+  const navigation = useNavigation();
```

Comment:
Missing a type for the navigation. I'd also recommend moving RootStackParamList to `App.tsx` or a common `types.ts` file so other screens can share it

```
type PhotoDetailNavigationProp = StackNavigationProp<
  RootStackParamList,
  'PhotoDetail'
>;

const PhotoDetailScreen = () => {
  const route = useRoute<PhotoDetailRouteProp>();
  const imageUrl = route.params.imageUrl;
  const navigation = useNavigation<PhotoDetailNavigationProp>();
```